src/sentry/receivers/releases.py

```diff
@@ -235,6 +243,13 @@ def save_release_activity(instance: Release, created: bool, **kwargs):
             )
 
 
+pre_save.connect(
+    validate_release_empty_version,
+    sender=Release,
+    dispatch_uid="validate_release_empty_version",
+    weak=False,
+)
+
```

Comment:
Not sure we should do this. It means that we'll raise an exception when creating a bad release. This will cause events with bad releases to fail in `save_event`, and will also cause `release_monitor` to stop processing when it fails to create the release.

We could potentially catch the validation error in these places and continue on

Reply:
I can definitely add try except blocks to the places where we call `Release.objects.create()` or `Release.get_or_create()`. Do you think the approach to block saving Releases with invalid version by throwing an error still makes sense though? I scanned the codebase and `ReleaseSerializer` goes through the same field validation as the `pre_save` approach:
https://github.com/getsentry/sentry/blob/45df001089840180249733b5da2c6f11302339f5/src/sentry/api/serializers/rest_framework/release.py#L75-L76

How catastrophic would it be if an uncaught `ValidationError` gets thrown in `save_event` or `release_monitor` processing? Would it completely block the whole event processing pipeline for all events, or result in just that event getting dropped?

To me, it seems like we definitely don't want to create releases with empty version and there was some effort to introduce validation in most Release-creation calls and stop creating those rows but some minor edge case fell through the cracks and it's still happening.

Reply:
If we failed like that in `save_event` it'd cause problems with saving the event. Basically everything after https://github.com/getsentry/sentry/blob/272d35503a2d5174dfa8cad57f94a2354e453bf3/src/sentry/event_manager.py#L497 would fail. I don't know the total implications here, but it feels like the event would be in an incomplete state. It shouldn't cause any processing backlogs though, since this is already running inside of a task.

In release monitor it'd break the task in the middle of processing. It isn't a severe breakage or anything because this task isn't extremely important, but we'd still like to keep it working. We probably should add a generic try/catch around each org we process here anyway, so that one specific org can't break the entire task.

I'm fine with catching the exceptions in these sensitive places, I think the approach makes sense.

Reply:
Yeah, the number of times these releases are getting created are fairly low. I can appreciate not wanting to save events in an inconsistent state, but I think this change is relatively low risk.

Is there anything else you think I need to add here before you're comfortable approving this PR?

Reply:
I think catching the exception should be enough to make things work